### PR TITLE
Changes to resolve build failures in wseb transport

### PR DESCRIPTION
**Pull request by cmebarrow**
The purpose of this PR is to address [Gateway Travis build failed in wseb transport, 3 test failures #424](https://github.com/kaazing/gateway/issues/424). Main changes are:

- Changed WsebConnector.createHandler to include values from the supported protocols option on the resource address in the X-WebSocket-Protocol header, in order to be consistent with WsnConnector. Altered wseb connector specification test OpeningIT to use that option to set multiple values for the header. Fixed setting of the X-WebSocket-Protocol and X-WebSocket-Extension headers in WsebConnector create session initializer to use a separate header line for each value, because it typifies the code and is more easily compatibible with the wse specification tests. Fixed validation of X-WebSocket-Protocol and X-WebSocket-Extensions header in WsebConnector.readHandler.doSessionClosed, including allowing for extension parameters.
- Changed WsebDownstreamHandler.sessionOpened not to abruptly close the downstream if the wsebSession is closing. This allows processing to continue into WsebSession.attachWriter0 which deals with the isClosing case, and is enhanced to send the CLOSE frame if not already sent. This should fix failures in TextIT.clientSendTextInvalidUTF8 involving the downstream being closed with the wrong content type and no close frame.
- Fixed WsebConnectorLoggingIT.shouldLogProtocolException to wait for the closed event to fire to make sure context.assertIsSatisfied does not race ahead of it.
- The failure in ProxiesIT is addressed by a fix made to its spec script in [k3po PR 307] (https://github.com/k3po/k3po/pull/307), available in k3po release 3.0.0-alpha-19.
- Removed a couple of unnecessary debug log messages from WsebDownstreamHandler and WsebUpstreamHandler (the exceptions concerned I logged by LoggingFilter anyway)
- Enhanced HttpConnector to call LoggingFilter.moveAfterCodec in create session initializer so log trace includes a legible form of the HTTP create request.

This PR also addresses #423 by using the argLine failsafe configuration parameter in gateway pom.xml to set the -Xmx1024m java option to limit the Java heap size.

### CODE REVIEW

**Comment by cmebarrow** [comment]
Travis build failed because of https://github.com/kaazing/gateway/issues/423. Restarted since that issue is sporadic.

File: transport/wseb/src/test/java/org/kaazing/gateway/transport/wseb/specification/wse/connector/OpeningIT.java [hunk 1]
```diff
@@ -334,7 +335,7 @@ public void shouldFailConnectionWhenXWebSocketExtensionsNotRequested() throws Ex
             }
         });
         Map<String, Object> connectOptions = new HashMap<String, Object>();
-        connectOptions.put("extensions", Arrays.asList("primary, secondary"));
+        connectOptions.put("extensions", Arrays.asList(new String[]{"primary, secondary"}));
```
Comment (jitsni) [inline transport/wseb/src/test/java/org/kaazing/gateway/transport/wseb/specification/wse/connector/OpeningIT.java]: *Two* String objects in List ?? then Arrays.asList("primary", "secondary")

File: transport/wseb/src/test/java/org/kaazing/gateway/transport/wseb/specification/wse/connector/OpeningIT.java [hunk 2]
```diff
@@ -133,7 +133,7 @@ public void shouldEstablishConnectionWithRequestHeaderXWebSocketExtensions() thr
             }
         });
         Map<String, Object> connectOptions = new HashMap<String, Object>();
-        connectOptions.put("extensions", Arrays.asList("primary, secondary"));
+        connectOptions.put("extensions", Arrays.asList(new String[]{"primary", "secondary"}));
```
Comment (jitsni) [inline transport/wseb/src/test/java/org/kaazing/gateway/transport/wseb/specification/wse/connector/OpeningIT.java]: explicit String[] is not needed.

**Comment by jitsni** [comment]
Looks good to me. Added some minor comments

File: transport/wseb/src/test/java/org/kaazing/gateway/transport/wseb/specification/wse/connector/OpeningIT.java (hunk 2 above)
Comment (cmebarrow) [inline transport/wseb/src/test/java/org/kaazing/gateway/transport/wseb/specification/wse/connector/OpeningIT.java]: Thanks, will fix. And yup, the options are different types :(

**Pull request merged**